noobaa/noobaa-core
# multiple fixes following failed s3 tests

### Explain the changes
small fixes for s3 tests:
1. add condition to check limit for number of deleted objects is larger than 1000 as specified in aws specification 
2. change lifecycle rule so it will be possible to have prefix instead of filter section
3. changed error returned for invalid encryption algorithm to InvalidArgument

### Testing Instructions:
run the following s3 tests:
1. s3tests_boto3.functional.test_s3:test_multi_object_delete_key_limit s3 test
2. s3tests_boto3.functional.test_s3:test_lifecycle_set_date
3. s3tests_boto3.functional.test_s3:test_put_obj_enc_conflict_bad_enc_kms
